src/transports/dash/segment_parser.ts

```diff
@@ -69,18 +71,40 @@ export default function parser({ content,
                                                             0),
                           appendWindow: [period.start, period.end] });
   } else { // it is an initialization segment
+    const shouldExtractCompleteInitChunk = segment.range === undefined;
+    const completeInitChunk = shouldExtractCompleteInitChunk ?
+      extractCompleteInitChunk(chunkData) : chunkData;
+
+    if (completeInitChunk === null &&
+        (
+          nextSegments === null ||
+          nextSegments.length === 0
+        )
+    ) {
+      if (!(representation.index instanceof BaseRepresentationIndex)) {
```

Comment:
The usual way of dealing with that type of problem is by using the `privateInfos` property